src/main/java/com/ibm/eventstreams/connect/jdbcsink/database/writer/JDBCWriter.java

```diff
@@ -50,107 +51,205 @@ private boolean doesTableExist(Connection connection, String tableName) throws S
+        return connection.getMetaData().getDatabaseProductName().toLowerCase().contains("db2");
+    }
+
+    private boolean isMySQL(Connection connection) throws SQLException {
+        return connection.getMetaData().getDatabaseProductName().toLowerCase().contains("mysql");
+    }
+
+    private String getPostgreSQLFieldType(Schema.Type fieldType) {
+        switch (fieldType) {
+            case INT8:
+            case INT16:
+            case INT32:
+                return "INTEGER";
+            case INT64:
+                return "BIGINT";
+            case FLOAT32:
+                return "REAL";
+            case FLOAT64:
+                return "DOUBLE PRECISION";
+            case BOOLEAN:
+                return "BOOLEAN";
+            case STRING:
+                return "VARCHAR(255)";
+            default:
+                return "VARCHAR(255)";
+        }
+    }
+
+    private String getDB2FieldType(Schema.Type fieldType) {
+        switch (fieldType) {
             case INT8:
             case INT16:
             case INT32:
                 return "INTEGER";
             case INT64:
                 return "BIGINT";
             case FLOAT32:
+                return "REAL";
             case FLOAT64:
+                return "DOUBLE";
+            case BOOLEAN:
+                return "BOOLEAN";
+            case STRING:
+                return "VARCHAR(255)";
+            default:
+                return "VARCHAR(255)";
+        }
+    }
+
+    private String getMySQLFieldType(Schema.Type fieldType) {
+        switch (fieldType) {
+            case INT8:
+            case INT16:
+            case INT32:
+                return "INT";
+            case INT64:
+                return "BIGINT";
+            case FLOAT32:
                 return "FLOAT";
+            case FLOAT64:
+                return "DOUBLE";
             case BOOLEAN:
                 return "BOOLEAN";
+            case STRING:
+                return "VARCHAR(255)";
             default:
                 return "VARCHAR(255)";
         }
     }
 
+    private String getIdColumnDefinition() {
+        try {
+            if (isPostgreSQL(dataSource.getConnection())) {
+                if (getPostgresMajorVersion() <= 9) {
+                    return "id SERIAL PRIMARY KEY";
+                } else {
+                    return "id INTEGER PRIMARY KEY GENERATED BY DEFAULT AS IDENTITY";
+                }
+            } else if (isDB2(dataSource.getConnection())) {
+                return "id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY";
+            } else if (isMySQL(dataSource.getConnection())) {
+                return "id INTEGER PRIMARY KEY AUTO_INCREMENT";
+            }
+        } catch (SQLException e) {
+            e.printStackTrace();
+        }
+        return "id INTEGER GENERATED BY DEFAULT AS IDENTITY PRIMARY KEY";
+    }
+
+    private int getPostgresMajorVersion() {
```

Comment:
possibly a config/connection handling class separately?

Reply:
should be separated ideally